Author: Andreas Beckmann <anbe@debian.org>
Description: fix some typos found by lintian

--- a/src/XF86Config-parser/Monitor.c
+++ b/src/XF86Config-parser/Monitor.c
@@ -379,7 +379,7 @@ xconfigParseVerboseMode (void)
             Error (UNEXPECTED_EOF_MSG, NULL);
             break;
         default:
-            Error ("Unexepcted token in verbose \"Mode\" entry\n", NULL);
+            Error ("Unexpected token in verbose \"Mode\" entry\n", NULL);
         }
     }
     if (!had_dotclock)
